src/main/java/seedu/address/model/event/StartTime.java

```diff
@@ -3,55 +3,55 @@
 import java.time.LocalTime;
-import java.time.format.DateTimeFormatter;
-import java.time.format.DateTimeParseException;
+
 
 /**
  * Represents an Event's starting time in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidStartTime(String)}
  */
 public class StartTime implements Comparable<StartTime> {
 
-    public static final String MESSAGE_CONSTRAINTS = "Start time must be in format: hh/mm/AM, hh/mm/PM or hh:mm";
-
-    //for checking if valid input date format
-    private static final DateTimeFormatter checkFormatter = DateTimeFormatter
-            .ofPattern("[HH:MM][hh/mm/a][h/mm/a][hh/m/a][h/m/a]");
-
-    //for changing to storage friendly format
-    private static final DateTimeFormatter logFormatter = DateTimeFormatter.ofPattern("hh/mm/a");
-
-    //for changing to user-readable format
-    private static final DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("hh:mm a");
-
+    public static final String MESSAGE_FORMAT_CONSTRAINTS = "Start time must be in format: hh:mm.";
+    public static final String MESSAGE_VALUE_CONSTRAINTS = "%s exceeds the range of valid time value.";
+    private static final DecimalFormat df = new DecimalFormat("00");
     public final LocalTime time;
 
 
     /**
      * Constructs a {@code StartTime}.
-     *
      * @param startTime A valid start time.
      */
     public StartTime(String startTime) {
         requireNonNull(startTime);
         assert !startTime.isBlank();
-        checkArgument(isValidStartTime(startTime), MESSAGE_CONSTRAINTS);
-        this.time = LocalTime.parse(startTime, checkFormatter);
+        checkArgument(isValidStartTime(startTime) != 0, MESSAGE_FORMAT_CONSTRAINTS);
+        checkArgument(isValidStartTime(startTime) != -1, String.format(MESSAGE_VALUE_CONSTRAINTS, startTime));
+        String [] parsedTime = startTime.split(":", 2);
+        this.time = LocalTime.of(Integer.parseInt(parsedTime[0]), Integer.parseInt(parsedTime[1]), 0);
     }
 
     /**
-     * Returns true if a given string is a valid StartTime input.
-     * @return boolean
+     * Checks if a given string is a valid StartTime input.
+     * @return int value, 0 represent invalid time format, -1 represents invalid time value, 1 represents valid time.
      */
-    //found from https://mkyong.com/java/how-to-check-if-date-is-valid-in-java/
-    public static boolean isValidStartTime(String test) {
+    public static int isValidStartTime(String test) {
         try {
-            LocalTime.parse(test, checkFormatter);
-        } catch (DateTimeParseException e) {
-            return false;
+            String [] parsedTime = test.split(":", 2);
+            Integer formatCheck = Integer.parseInt(parsedTime[0]) + Integer.parseInt(parsedTime[1]);
+            boolean lengthCheck = parsedTime[0].length() == 2 && parsedTime[1].length() == 2;
+            LocalTime.of(Integer.parseInt(parsedTime[0]), Integer.parseInt(parsedTime[1]), 0);
```

Comment:
Same as above `LocalTime.of()` line

```suggestion
            LocalTime.of(Integer.parseInt(parsedTime[0]), Integer.parseInt(parsedTime[1]));
```